discovery-infra/tests/test_validations.py

```diff
@@ -0,0 +1,76 @@
+
+
+class TestValidations(BaseTest):
+    @pytest.mark.regression
+    def test_basic_cluster_validations(self, api_client, cluster):
+        cluster_id = cluster().id
+
+        # Check initial validations
+        self.wait_until_cluster_is_pending_for_input(cluster_id, api_client)
+        orig_cluster = api_client.cluster_get(cluster_id)
+        self.assert_cluster_validation(orig_cluster, "configuration", "pull-secret-set", "success")
+        self.assert_cluster_validation(orig_cluster, "hosts-data", "all-hosts-are-ready-to-install", "success")
+        self.assert_cluster_validation(orig_cluster, "hosts-data", "sufficient-masters-count", "failure")
+        self.assert_cluster_validation(orig_cluster, "network", "machine-cidr-defined", "failure")
+        self.assert_cluster_validation(orig_cluster, "network", "machine-cidr-equals-to-calculated-cidr", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "api-vip-defined", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "api-vip-valid", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-defined", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-valid", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "dns-domain-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "cluster-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "service-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "no-cidrs-overlapping", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "network-prefix-valid", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "ntp-server-configured", "success")
+
+        # Check base DNS domain
+        api_client.update_cluster(cluster_id, {"base_dns_domain": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["failure"])
+        api_client.update_cluster(cluster_id, {"base_dns_domain": orig_cluster.base_dns_domain})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["success"])
+
+        # Check pull secret
+        api_client.update_cluster(cluster_id, {"pull_secret": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["failure"])
+        api_client.update_cluster(cluster_id, {"pull_secret": env_variables['pull_secret']})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["success"])
+
+    def test_host_insufficient(self, api_client, cluster, nodes):
+        cluster_id = cluster().id
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+
+        # Modify vCPU count on the first node to be insufficient
```

Comment:
is it possible to replace comments with logs?

Reply:
all the methods that being called have logs, why do we need it here?

Reply:
Even if the getter methods have logs (which I'm not sure of it), logs on the test level should explain the macro of what the test expects. Something like ``"choosing randomly host and change its CPU cores number to be non-sufficient as a master"``
The implementation logs should be more implementation-driven, to give details in a case of a bug
I also would like our test logs to be of level of info and implementation to have debug level, but it's a more major change

Reply:
in my POV it is bad practice to log too much, and to log each step in the test level sounds irrelevant as in any case there is a need to understand test purpose, by reading its code (and comments) or a written TP.

Reply:
It's fine, we can agree to disagree :)
But I'm sure that if we could have the framework to report on the major things it would benefit us. Probably more on the more complex tests than on the simple ones

Reply:
Agree :)